x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -1501,16 +1484,28 @@ export class RulesClient {
       await this.unsecuredSavedObjectsClient.update('alert', id, {
         scheduledTaskId: scheduledTask.id,
       });
-      if (apiKeyToInvalidate) {
-        await markApiKeyForInvalidation(
-          { apiKey: apiKeyToInvalidate },
-          this.logger,
-          this.unsecuredSavedObjectsClient
-        );
-      }
     }
   }
 
+  private async createNewAPIKeySet({
+    attributes,
+    username,
+  }: {
+    attributes: RawRule;
+    username: string | null;
+  }): Promise<Pick<RawRule, 'apiKey' | 'apiKeyOwner'>> {
+    let createdAPIKey = null;
+    try {
+      createdAPIKey = await this.createAPIKey(
+        this.generateAPIKeyName(attributes.alertTypeId, attributes.name)
+      );
+    } catch (error) {
+      throw Boom.badRequest(`Error enabling rule: could not create API key - ${error.message}`);
```

Comment:
**question:** should we remove references of the calling function incase this gets re-useed later on?
```suggestion
      throw Boom.badRequest(`Error creating API key for rule: ${error.message}`);
```

Reply:
Should we unwrap this fro try-catch or fix the value? :)

Reply:
My opinion would be to keep the try-catch. It provides a bit more context as to where this error comes from when we don't have a stack trace with it.

Reply:
done